node/tests/swamp/swamp.go

```diff
@@ -226,3 +235,37 @@ func (s *Swamp) NewLightNodeWithStore(store node.Store, options ...node.Option)
 
 	return node
 }
+
+// RemoveNode removes a node from the swamp's node slice
+// this allows reusage of the same var in the test scenario
+// if the user needs to stop and start the same node
+func (s *Swamp) RemoveNode(n *node.Node, t node.Type) error {
+	switch t {
+	case node.Light:
+		return s.remove(n, s.LightNodes)
+	case node.Bridge:
+		return s.remove(n, s.BridgeNodes)
+	default:
+		return fmt.Errorf("no such type or node")
+	}
+}
+
+func (s *Swamp) remove(rn *node.Node, sn []*node.Node) error {
+	if len(sn) == 1 {
+		sn = nil
+		return nil
+	}
+
+	initSize := len(sn)
+	for i := 0; i < len(sn); i++ {
+		if sn[i] == rn {
+			sn = append(sn[:i], sn[i+1:]...)
```

Comment:
Here we have a bug. Append in this case will create a new slice with a newly allocated memory behind it and you only set it to `sn` in the local stack, instead, you should set the result here directly into Swamp.LightNode or Swamp.BridgeNodes. 

Ping @vgonkivs for a more in-depth explanation.

Reply:
@Bidon15 https://github.com/golang/go/wiki/SliceTricks

Reply:
done according to ideas above. Thanks!